### DISCUSSION TITLE
[RFC] Alternative storage backend

### DISCUSSION THREAD

**Discussion by mamazu** [phpactor/phpactor#2232] [Ideas]
Currently phpactor only supports serializing records to php strings, which is a very verbose format so an index can easily be 600MB for a mid sized project.

* Have we considered using json or maybe a database to represent the index?
* How easy / difficult do you think it is going to be?

**Comment by dantleech** [comment]
- JSON should be slower (if that matters I don't know)
- SQLite is the other option, but not everybody has SQLite.
- Supporting 2 backends would probably coincide with wider refactoring, so probably not trivial.

But yes, I would like to add support for SQLite at some point.

**Reply by aharabara** [reply]
As a  quick to implement solution we could use https://github.com/typhoon-php/opcache.
It would compile objects into code and would benefit from opcache performance.
@dantleech  what are your thought about on this topic?
